federation/pkg/federation-controller/replicaset/planner/planner.go

```diff
@@ -39,9 +41,9 @@ type byWeight []*namedClusterReplicaSetPreferences
 func (a byWeight) Len() int      { return len(a) }
 func (a byWeight) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
 
-// Preferences are sorted according by decreasing weight and increasing clusterName.
+// Preferences are sorted according by decreasing weight and increasing hash (built on top of cluster name and rs name).
```

Comment:
Actually - I thought that I understood this change initially, but now I no longer think I understand it.

Please add a comment, why you actually can't compare based on clusterName here.